gui/wxpython/datacatalog/catalog.py

```diff
@@ -46,21 +51,33 @@ def __init__(self, parent, giface=None, id=wx.ID_ANY,
         self.tree = DataCatalogTree(self, giface=giface)
         self.tree.showNotification.connect(self.showNotification)
 
+        # infobar for data catalog
+        self.infoBar = InfoBar(self)
+
+        # infobar manager for data catalog
+        self.infoManager = DataCatalogInfoManager(self,
+                                                  infobar=self.infoBar,
+                                                  giface=self.giface)
         # some layout
         self._layout()
 
     def _layout(self):
         """Do layout"""
-        sizer = wx.BoxSizer(wx.VERTICAL)
-
-        sizer.Add(self.toolbar, proportion=0,
-                  flag=wx.EXPAND)
-
-        sizer.Add(self.tree.GetControl(), proportion=1,
-                  flag=wx.EXPAND)
+        self.sizer = wx.BoxSizer(wx.VERTICAL)
+        self.sizer.Add(self.toolbar, proportion=0, flag=wx.EXPAND)
+        self.sizer.Add(self.tree.GetControl(), proportion=1, flag=wx.EXPAND)
+
+        # Show first infobar for first-time user
+        if is_current_mapset_in_demolocation():
+            self.sizer.Insert(1, self.infoBar,
+                              proportion=0, flag=wx.EXPAND)
+            buttons1 = [("Create new Location", self.OnCreateLocation),
+                        ("Learn More", self.infoManager._onLearnMore)]
+            self.infoManager.ShowDataStructureInfo(buttons1)
```

Comment:
This needs to go at the end of constructor and it needs to be called with wx.CallAfter, that should fix the layout issue.